server/api/submission_handler.py

```diff
@@ -10,20 +10,21 @@
 
 @submission_handler.route('/contestImage/submission/<contest_id>', methods=['POST'])
 def create_submission(contest_id):
-    try:
-        s3.upload_fileobj(request.files['file'], S3_BUCKET, request.form["file_name"])
-    except Exception as e:
-        return jsonify({'error': "file not found"})
-
     token = request.cookies.get("auth_token")
 
     if token is None:
         return jsonify({"error": "auth_token missing"}), 401
 
     data = jwt.decode(token, app.app.config['JWT_SECRET'], algorithms=['HS256'])
     current_user = User.query.filter_by(username=data['username']).first()
-
+    
     image = 'https://'+S3_BUCKET+'.s3.amazonaws.com/'+request.form["file_name"]
+
+    try:
+        s3.upload_fileobj(request.files['file'], S3_BUCKET, 'userid_{}/{}'.format(current_user.id, request.form["file_name"]))
```

Comment:
I think we can use a sub-directory for different submissions. In case the user submits to different contests with a generic file name like file1.png. 

Also, I think it's better to have the contest_id as the parent folder that contains different users submission, directory structures like /root/contest_id/user_id/files.

```suggestion
        s3.upload_fileobj(request.files['file'], S3_BUCKET, 'contest_id_{}/user_id_{}/{}'.format(contest_id, current_user.id, request.form["file_name"]))
```
This way it is also potentially easier for all submissions to be loaded under the same contest. 


Or we could use the submission_id instead, and properly create a one-to-many relationship between contest and submission later on. 

```suggestion
        s3.upload_fileobj(request.files['file'], S3_BUCKET, 'submission_id_{}/{}'.format(new_submission.id, request.form["file_name"]))
```

this one requires you to create a row in the database before uploading to S3.

Reply:
I personally prefer the second one since it is probably easier to implement features from the user's perspective.

Reply:
one additional thing if we want ot take it to the next level is to append a timestamp to the file too so that would ensure it will always be unique